extensions/vscode/src/utils/window.test.ts

```diff
@@ -0,0 +1,141 @@
+// Copyright (C) 2025 by Posit Software, PBC.
+
+import { describe, expect, beforeEach, test, vi } from "vitest";
+import { window } from "vscode";
+import {
+  showErrorMessageWithTroubleshoot,
+  showInformationMsg,
+  taskWithProgressMsg,
+  runTerminalCommand,
+} from "./window";
+
+const terminalMock = {
+  sendText: vi.fn(),
+  show: vi.fn(),
+  exitStatus: {
+    code: 0,
+  },
+};
+
+vi.mock("vscode", () => {
+  // mock Disposable
+  const disposableMock = vi.fn();
+  disposableMock.prototype.dispose = vi.fn();
+
+  // mock window
+  const windowMock = {
+    showErrorMessage: vi.fn(),
+    showWarningMessage: vi.fn(),
+    showInformationMessage: vi.fn(),
+    withProgress: vi.fn().mockImplementation((_, progressCallback) => {
+      progressCallback();
+    }),
+    createTerminal: vi.fn().mockImplementation(() => {
+      terminalMock.sendText = vi.fn();
+      terminalMock.show = vi.fn();
+      return terminalMock;
+    }),
+    onDidCloseTerminal: vi.fn().mockImplementation((fn) => {
+      setTimeout(() => fn(terminalMock), 100);
+      return new disposableMock();
+    }),
+  };
+
+  return {
+    Disposable: disposableMock,
+    window: windowMock,
+    ProgressLocation: {
+      SourceControl: 1,
+      Window: 10,
+      Notification: 15,
+    },
+  };
+});
+
+describe("Consumers of vscode window", () => {
+  beforeEach(() => {
+    terminalMock.exitStatus.code = 0;
+  });
+
+  test("showErrorMessageWithTroubleshoot", () => {
+    showErrorMessageWithTroubleshoot("Bad things happened");
+    expect(window.showErrorMessage).toHaveBeenCalledWith(
+      "Bad things happened. See [Troubleshooting docs](https://github.com/posit-dev/publisher/blob/main/docs/troubleshooting.md) for help.",
+    );
```

Comment:
Thanks for adding a test here. I should have done it with the introduction of the function 😅